src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -180,22 +180,11 @@ Namespace Microsoft.CodeAnalysis.Operations
                     Dim lastParameterIndex = parameters.Length - 1
                     Dim kind As ArgumentKind = ArgumentKind.Explicit
 
-                    If argument.WasCompilerGenerated AndAlso Not invocationWasCompilerGenerated Then
-
-                        If index = lastParameterIndex AndAlso ParameterIsParamArray(parameters(lastParameterIndex)) Then
```

Comment:
>index = lastParameterIndex [](start = 27, length = 26)

It looks like this check is dropped, this doesn't look safe.

Reply:
The new check is precisely the check used by other sections of the compiler to determine if the parameter is a paramarray. I've checked all instances of creating and updating BoundArrayCreation, and the only time that flag is set is here: [https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb#L2705](https://github.com/dotnet/roslyn/blob/master/src/Compilers/VisualBasic/Portable/Binding/Binder_Invocation.vb#L2705).

---
In reply to: [190111038](https://github.com/dotnet/roslyn/pull/26781#discussion_r190111038) [](ancestors = 190111038)

Reply:
I can reintroduce the index check if you want, but this is what `BoudnArrayCreation.IsParamArrayArgument` is intended to convey.

---
In reply to: [190316564](https://github.com/dotnet/roslyn/pull/26781#discussion_r190316564) [](ancestors = 190316564,190111038)

Reply:
Talked with Aleksey offline, this check is sufficient.

---
In reply to: [190316977](https://github.com/dotnet/roslyn/pull/26781#discussion_r190316977) [](ancestors = 190316977,190316564,190111038)